### PR TITLE
Make OpenWrt more SPDX compliant

### PR DESCRIPTION
**Pull request by aparcar**
The license folder is a core part of OpenWrt and all GPL-2.0 licensed.
Adding a SPDX license tag allows machines to check licenses.

The `(C)` is optional based on the GPS 2.0 text, so why waste 3 bytes on
every text file. Update copyright years to 2020, this should be done in
a recurring manner, e.g. yearly.

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by aparcar** [comment]
# SUMMARY

* Bad licenses:
* Deprecated licenses:
* Licenses without file extension:
* Missing licenses: GPL-2.0, GPL-2.0+
* Unused licenses:
* Used licenses: BSD-2-Clause, BSD-3-Clause, GPL-1.0-or-later, GPL-2.0-only, GPL-2.0-or-later, ISC, Linux-syscall-note, MIT
* Read errors: 1
* Files with copyright information: 2082 / 9920
* Files with license information: 973 / 9920

File: include/autotools.mk [hunk 1]
```diff
@@ -1,9 +1,6 @@
+# Copyright 2007-2020 OpenWrt.org
 #
-# Copyright (C) 2007-2012 OpenWrt.org
-#
-# This is free software, licensed under the GNU General Public License v2.
-# See /LICENSE for more information.
-#
+# SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (adschm) [inline include/autotools.mk:3]: I think the SPDX license should be in the first line, and the copyright may come afterwards. That's also what the kernel recommendations say:

[...]
The SPDX license identifier in kernel files shall be added at the first possible line in a file which can contain a comment. For the majority or files this is the first line, except for scripts which require the ‘#!PATH_TO_INTERPRETER’ in the first line. For those scripts the SPDX identifier goes into the second line.
[...]

I'm sorry for creating additional work by this ...

File: include/depends.mk [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 # define a dependency on a subtree
 # parameters:
-#	1: directories/files
-#	2: directory dependency
-#	3: tempfile for file listings
-#	4: find options
+# 	1: directories/files
+# 	2: directory dependency
+# 	3: tempfile for file listings
+# 	4: find options
```
Comment (adschm) [inline include/depends.mk:10]: What's that?

File: LICENSE [hunk 3]
```diff
@@ -1,339 +0,0 @@
-                    GNU GENERAL PUBLIC LICENSE
```
Comment (adschm) [inline LICENSE:1]: Like for README, please check whether removing this file will break anything ...

File: include/autotools.mk (hunk 1 above)
Comment (aparcar) [reply include/autotools.mk:3]: Yea I remember that too, wonder why the `reuse` tools doesn't follow this idea. I'll patch it

**Comment by adschm** [comment]
I just pushed the scripts string update and did the target update myself (without touching patches, though).

File: include/depends.mk (hunk 2 above)
Comment (aparcar) [reply include/depends.mk]: Fixed

File: include/autotools.mk [hunk 4]
```diff
@@ -1,9 +1,6 @@
...
 #
-# Copyright (C) 2007-2012 OpenWrt.org
-#
-# This is free software, licensed under the GNU General Public License v2.
```
Comment (adschm) [inline include/autotools.mk:4]: Here (and below) you are converting GPL-2.0 to GPL-2.0-or-later?

**Comment by adschm** [comment]
please don't use spdx as prefix, but refer to the folders/code hierarchy as for other patches ...

File: target/linux/ath79/patches-4.19/0028-MIPS-ath79-drop-machfiles.patch [hunk 5]
```diff
@@ -42,7 +42,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 --- a/arch/mips/ath79/Kconfig
 +++ b/arch/mips/ath79/Kconfig
 @@ -1,79 +1,6 @@
- # SPDX-License-Identifier: GPL-2.0
+ # SPDX-License-Identifier: GPL-2.0-only
```
Comment (adschm) [inline target/linux/ath79/patches-4.19/0028-MIPS-ath79-drop-machfiles.patch:45]: This line is from kernel, we don't add/remove it, but just use it to determine the position of the changes. Lines like this must not be changed.

Comment (adschm) [reply target/linux/ath79/patches-4.19/0028-MIPS-ath79-drop-machfiles.patch:45]: And personally I'm not sure whether we should touch the patches at all.

Comment (aparcar) [reply target/linux/ath79/patches-4.19/0028-MIPS-ath79-drop-machfiles.patch]: I removed the target patch for now.

**Comment by aparcar** [comment]
> please don't use spdx as prefix, but refer to the folders/code hierarchy as for other patches ...

Done

**Comment by neheb** [comment]
Relevant: https://github.com/systemd/systemd/commit/810adae9e9e0b028bd1d210918d0380e15c5595a

**Comment by aparcar** [comment]
Thanks! However I was planing to update to a more modern Standard https://emojipedia.org/copyright/

**Comment by aparcar** [comment]
> This important patch is very important.

;)

**Comment by adschm** [comment]
I've personally never understood why people use "Copyright (C)" or similar as it's redundant to me. One also don't say it's 30 kilograms (kg) ...
However, as it's essentially completely irrelevant, I'd only _change_ it if the new variant is superior for some objective reason, or we really follow a common standard afterwards.

**Comment by aparcar** [comment]
GPL says it's optional, let's make it the standard.

**Comment by f00b4r0** [comment]
IANAL but ISTR that the "(C)" is actually required in some legislations. i.e. that's not a license thing, that's a law one, and I'd thread very carefully there. IMHO the size argument cuts both ways.

**Comment by aparcar** [comment]
Well the GPL text says it's optional, but I don't really care. @adschm preferences?

**Comment by adschm** [comment]
As stated before, I wouldn't do a purely "cosmetic" change without a good reason.

**Comment by adschm** [comment]
Looking at this again, I'm still not convinced we should add all those license headers where they have been missing so far.

If we can deduce the type of the files' licenses from our "project license", I don't see a point of adding them to the files, as we know the license already. And if we cannot deduce anything, we cannot add anything.

So, I'd really just limit this to a mere replacement of long text licenses to license identifiers/headers, and that's it.

**Comment by aparcar** [comment]
@adschm I updated the files so we use spdx where there was already a license and keep the unlicensed files as before. I'll write the SFC a email asking what to do in such cases.

**Comment by aparcar** [comment]
@adschm I wrote the SFC an email and added more files from `config/`, please merge this and once SFC has shared an opinion I'll do some follow ups

**Comment by bkuhn** [comment]
I sent an email about this to @aparcar last week.  This issue is complicated and we should probably have an email thread about it that will be easier to discuss these complex issues here.  @aparcar, feel free to add to the Cc list anyone you think should be included in that thread (e.g., @olek2).

File: LICENSES/BSD-2-Clause.txt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Copyright (c) <year> <owner>. All rights reserved.
```
Comment (CodeFetch) [inline LICENSES/BSD-2-Clause.txt:1]: Shouldn't there be a year given?

Comment (aparcar) [reply LICENSES/BSD-2-Clause.txt:1]: From my understanding that's the license text and year plus text come from the individual files.

**Comment by aparcar** [comment]
@adschm please merge the second commit

**Comment by adschm** [comment]
I have reshaped this a little and pasted it here:

https://patchwork.ozlabs.org/project/openwrt/list/?series=229295

**Pull request closed without merging**